tests/trestle/core/models/write_file_action_test.py

```diff
@@ -0,0 +1,98 @@
+# -*- mode:python; coding:utf-8 -*-
+
+# Copyright (c) 2020 IBM Corp. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for trestle write to file actions class."""
+
+import pathlib
+
+from tests import test_utils
+
+from trestle.core.err import TrestleError
+from trestle.core.models.actions import FileContentType, WriteFileAction
+from trestle.core.models.elements import Element
+
+
+def test_write_file_action_yaml(tmp_yaml_file: pathlib.Path, sample_target):
+    """Test write yaml action."""
+    element = Element(sample_target)
+
+    try:
+        # string path should error
+        wa = WriteFileAction(tmp_yaml_file.as_posix(), element, FileContentType.YAML)
```

Comment:
This is more a question for the WriteFIleAction - but by passing extensions explicitly rather than implicitly we now have all of the errors you dealt with below. Wouldn't it be easier to infer typing and fail when it is not one of the valid types?